### PR TITLE
 Geodata in create and get emotly:  Modified rest api in order to accept in input and return as output the new geodata

### PR DESCRIPTION
**Pull request by ggc87**

### CODE REVIEW

File: emotly/controllers/emotly_controller.py [hunk 1]
```diff
@@ -42,7 +42,11 @@
 @require_https
 def list_emotlies():
     try:
-        emotlies = Emotly.objects.only("user", "created_at", "mood").all()
+        emotlies = \
```
Comment (michelangelo) [inline emotly/controllers/emotly_controller.py]: Why, really?:)

File: emotly/tests/test_rest_api.py [hunk 2]
```diff
@@ -29,17 +29,25 @@
...
-from emotly.models import User, Emotly, MOOD
+from emotly.models import User, Emotly, MOOD, Location
 from emotly.utils import generate_jwt_token
 
 
 # Tests for Emotly REST API.
 class RESTAPITestCase(unittest.TestCase):
     NUM_ELEMENT_2 = 2
     NUM_ELEMENT_3 = 3
+    u = None
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Do you need this?

File: emotly/tests/test_rest_api.py [hunk 3]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
                           base_url='https://localhost')
         self.assertEqual(rv.status_code, 200)
 
+    # Valid Geodata
+    def test_post_emotly_with_complete_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'coord': [1, 1],
+            'accuracy': 0.2,
+            'location_name': 'Pizzeria da Ciro'}}
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Block indentation.

File: emotly/tests/test_rest_api.py [hunk 4]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
...
+        rv = self.app.post(CONSTANTS.REST_API_PREFIX + "/emotlies/new",
+                           headers=headers, data=json.dumps(m),
+                           base_url='https://localhost')
+        self.assertEqual(rv.status_code, 200)
+
+    # Invalid Geodata.
+    def test_post_emotly_with_invalid_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'accuracy': 0.2,
+            'location_name': 'Pizzeria da Ciro'}}
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Ditto.

File: emotly/tests/test_rest_api.py [hunk 5]
```diff
@@ -485,7 +537,7 @@ def test_response_get_emotly(self):
         data = json.loads(rv.data.decode('utf-8'))
         # Number of key element in the json dictionary returned
         # by the rest api method
-        self.assertEqual(len(data["emotly"].keys()), self.NUM_ELEMENT_2)
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Good catch. If `NUM_ELEMENTS_2` is not being used anywhere else you can actually remove it altogether.

**Comment by michelangelo** [comment]
Looks good; just please fix those few simple remarks.

Also, please note that a commit message like this provides very little help in the history: its length and format makes it unreadable.
It'd be very appreciated if you could follow with the usual 50/72 format.

File: emotly/tests/test_rest_api.py [hunk 6]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
...
+        headers = {'content-type': 'application/json',
+                   'X-Emotly-Auth-Token': token}
+        rv = self.app.post(CONSTANTS.REST_API_PREFIX + "/emotlies/new",
+                           headers=headers, data=json.dumps(m),
+                           base_url='https://localhost')
+        self.assertEqual(rv.status_code, 200)
+
+    def test_post_emotly_with_incomplete_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'coord': [1, 1]}}
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Ditto.

File: emotly/tests/test_rest_api.py [hunk 7]
```diff
@@ -29,17 +29,25 @@
...
 class RESTAPITestCase(unittest.TestCase):
     NUM_ELEMENT_2 = 2
     NUM_ELEMENT_3 = 3
+    u = None
 
     def setUp(self):
         self.app = app.test_client()
+        self.u = User(nickname='testpost',
+                      email='test_post_emotly@example.com',
+                      password="FakeUserPassword123",
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Single or double quote is fine, just be consistent.:)

File: emotly/tests/test_rest_api.py [hunk 8]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
...
+    # Valid Geodata
+    def test_post_emotly_with_complete_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'coord': [1, 1],
+            'accuracy': 0.2,
+            'location_name': 'Pizzeria da Ciro'}}
+        token = generate_jwt_token(self.u)
+        headers = {'content-type': 'application/json',
+                   'X-Emotly-Auth-Token': token}
+        rv = self.app.post(CONSTANTS.REST_API_PREFIX + "/emotlies/new",
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Ditto (single/double quote).

File: emotly/tests/test_rest_api.py [hunk 9]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
...
+                           base_url='https://localhost')
+        self.assertEqual(rv.status_code, 200)
+
+    def test_post_emotly_with_incomplete_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'coord': [1, 1]}}
+        token = generate_jwt_token(self.u)
+        headers = {'content-type': 'application/json',
+                   'X-Emotly-Auth-Token': token}
+        rv = self.app.post(CONSTANTS.REST_API_PREFIX + "/emotlies/new",
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Ditto (single/double quote).

File: emotly/tests/test_rest_api.py [hunk 10]
```diff
@@ -261,6 +262,57 @@ def test_get_empty_own_list_emotlies(self):
...
+
+    # Invalid Geodata.
+    def test_post_emotly_with_invalid_geodata(self):
+        m = {'mood': 1, 'geodata': {
+            'accuracy': 0.2,
+            'location_name': 'Pizzeria da Ciro'}}
+        token = generate_jwt_token(self.u)
+        headers = {'content-type': 'application/json',
+                   'X-Emotly-Auth-Token': token}
+        rv = self.app.post(CONSTANTS.REST_API_PREFIX + "/emotlies/new",
```
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: Ditto (single/double quote).

File: emotly/tests/test_rest_api.py (hunk 7 above)
Comment (ggc87) [inline emotly/tests/test_rest_api.py]: I just copy pasted some piece of code,  it was already in there.  I didn't notice I'll check for other inconsistencies. ;)

File: emotly/tests/test_rest_api.py (hunk 8 above)
Comment (ggc87) [inline emotly/tests/test_rest_api.py]: Same here.

File: emotly/tests/test_rest_api.py (hunk 2 above)
Comment (ggc87) [inline emotly/tests/test_rest_api.py]: I need to declare a global user,  it is Initialized in the setup process.  Doing that we don't need to create a new user in each test case.

File: emotly/tests/test_rest_api.py (hunk 7 above)
Comment (michelangelo) [inline emotly/tests/test_rest_api.py]: That'd be great but not really needed: as we're gonna need to sweep the code anyway in the near future - if you want - you can just limit yourself to your own code for now.

**Comment by ggc87** [comment]
Comments fixed.

**Comment by michelangelo** [comment]
Amazing, @ggc87. Thanks!
Looks great!

**Pull request merged**